Source/JavaScriptCore/runtime/TemporalPlainYearMonth.cpp

```diff
@@ -118,6 +118,105 @@ String TemporalPlainYearMonth::toString(JSGlobalObject* globalObject, JSValue op
+        RETURN_IF_EXCEPTION(scope, { });
+        RELEASE_AND_RETURN(scope, result);
+    }
+
+    std::optional<JSObject*> options;
+    if (optionsValue) {
+        options = intlGetOptionsObject(globalObject, optionsValue.value());
+        RETURN_IF_EXCEPTION(scope, { });
+    }
+
+    if (isString) {
+        // See step 11 of ToTemporalYearMonth
+        // Overflow has to be validated even though it's not used,
+        // so that an error can be thrown for a bad overflow option)
+        if (options) {
+            toTemporalOverflow(globalObject, options.value());
+            RETURN_IF_EXCEPTION(scope, { });
+        }
+        RELEASE_AND_RETURN(scope, { });
+    }
+
+    if (itemValue.isObject()) {
+
+        if (itemValue.inherits<TemporalPlainYearMonth>())
+            return jsCast<TemporalPlainYearMonth*>(itemValue);
+
+        if (itemValue.inherits<TemporalPlainYearMonth>())
+            return TemporalPlainYearMonth::create(vm, globalObject->plainYearMonthStructure(), jsCast<TemporalPlainYearMonth*>(itemValue)->plainYearMonth());
+
+        JSObject* calendar = TemporalCalendar::getTemporalCalendarWithISODefault(globalObject, itemValue);
+        RETURN_IF_EXCEPTION(scope, { });
+
+        // FIXME: Implement after fleshing out Temporal.Calendar.
+        if (!calendar->inherits<TemporalCalendar>() || !jsCast<TemporalCalendar*>(calendar)->isISO8601()) [[unlikely]] {
+            throwRangeError(globalObject, scope, "unimplemented: from non-ISO8601 calendar"_s);
+            return { };
+        }
+
+        Variant<JSObject*, TemporalOverflow> optionsOrOverflow = TemporalOverflow::Constrain;
+        if (options)
+            optionsOrOverflow = options.value();
+        auto overflow = TemporalOverflow::Constrain;
+        auto plainYearMonth = TemporalCalendar::isoDateFromFields(globalObject, asObject(itemValue), TemporalDateFormat::YearMonth, optionsOrOverflow, overflow);
+        RETURN_IF_EXCEPTION(scope, { });
+
+        return TemporalPlainYearMonth::create(vm, globalObject->plainYearMonthStructure(), WTFMove(plainYearMonth));
+    }
+
+    throwTypeError(globalObject, scope, "can only convert to PlainYearMonth from object or string values"_s);
+    return { };
+}
+
+// https://tc39.es/proposal-temporal/#sec-temporal.plainyearmonth.from
+TemporalPlainYearMonth* TemporalPlainYearMonth::from(JSGlobalObject* globalObject, WTF::String string)
+{
+    VM& vm = globalObject->vm();
+    auto scope = DECLARE_THROW_SCOPE(vm);
+
+    // https://tc39.es/proposal-temporal/#sec-temporal-parsetemporaldatestring
+    // TemporalDateString :
+    //     CalendarDateTime
+    auto dateTime = ISO8601::parseCalendarDateTime(string, TemporalDateFormat::YearMonth);
+    if (dateTime) [[likely]] {
+        auto [plainDate, plainTimeOptional, timeZoneOptional, calendarOptional] = WTFMove(dateTime.value());
+        if (calendarOptional && StringView(calendarOptional.value()) != String::fromLatin1("iso8601")) [[unlikely]] {
```

Comment:
Please do not use `String::fromLatin1` for ASCII literals. We need to remove this from existing code and stop adding it in new code!
```suggestion
        if (calendarOptional && calendarOptional.value() != "iso8601"_s) [[unlikely]] {
```

Reply:
Happy to take out the `fromLatin1`, but it won't compile without the `StringView` (`calendarOptional.value()` is a `Vector<Latin1Character, maxCalendarLength>`).

Reply:
Best idiom for comparing a `Vector` of `Latin1Character` with a string constant is to call the `equal` function that compares two spans, something like:
```suggestion
        if (calendarOptional && !equal(calendarOptional->span(), "iso8601"_span8)) [[unlikely]] {
```
Using `StringView` brings branches for handling of 16-bit characters into play, unnecessarily.

Reply:
Done.